### PR TITLE
ListPropertiesEditing should not crash if there was a list in an image caption element

### LINKED ISSUE #13858: Crash when loading an image caption with a list inside (ListProperties)
## 📝 Provide detailed reproduction steps (if any)

1.  Go to https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/features/lists/lists.html#demo-2
2.  Set the following data

```html
<figure class="image"><img src="https://33333.cdn.cke-cs.com/rc1DFuFpHqcR3Mah6y0e/images/93c337f3c2eb83f436edba7e778a5983d212dd8ab79859d7.png"  width="1208"><figcaption><ul><li>foo</li></ul></figcaption></figure>
```

### ✔️ Expected result

_Silence._

### ❌ Actual result

```
spector.js:2 Uncaught TypeError: Cannot read properties of null (reading 'start')
    at e.on.priority (snippet.js:4:605515)
    at Ec.fire (snippet.js:4:5541)
    at Ec._convertItem (snippet.js:4:231683)
    at Ec._convertChildren (snippet.js:4:232140)
    at Object.convertChildren (snippet.js:4:230260)
    at Pc.upcastDispatcher.on.priority (snippet.js:4:236015)
    at Ec.fire (snippet.js:4:5541)
    at Ec._convertItem (snippet.js:4:231683)
    at Ec._convertChildren (snippet.js:4:232140)
    at Object.convertChildren (snippet.js:4:230260)
```

### ❓ Possible solution

List properties upcast converter tries to upcast attributes despite the "parent" list element not being upcasted in the first place (e.g. disallowed in the image caption). We need to check whether the model range exists first in the converter.

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by oleq**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Fix (list): `ListPropertiesEditing` should not crash if there was a list in an image caption element. Closes #13858.

### CODE REVIEW

File: packages/ckeditor5-list/src/listproperties/listpropertiesediting.ts [hunk 1]
```diff
@@ -372,6 +372,11 @@ function createAttributeStrategies( enabledProperties: ListPropertiesConfig ) {
 function upcastListItemAttributes( attributeStrategies: Array<AttributeStrategy> ) {
 	return ( dispatcher: UpcastDispatcher ) => {
 		dispatcher.on<UpcastElementEvent>( 'element:li', ( evt, data, conversionApi ) => {
+			// https://github.com/ckeditor/ckeditor5/issues/13858
+			if ( !data.modelRange ) {
+				return;
+			}
+
 			const listParent = data.viewItem.parent as ViewElement;
 			const listItem = data.modelRange!.start.nodeAfter || data.modelRange!.end.nodeBefore;
```
Comment (mlewand) [inline packages/ckeditor5-list/src/listproperties/listpropertiesediting.ts:381]: Nitpick: if I'm not mistaken your change means we can turn both `data.modelRange!` to `data.modelRange` now.

**Review by mlewand — APPROVED** [review]
LGTM, but I'll leave it for the team Core in case they want to do extra review.

File: packages/ckeditor5-list/src/listproperties/listpropertiesediting.ts (hunk 1 above)
Comment (arkflpc) [reply packages/ckeditor5-list/src/listproperties/listpropertiesediting.ts:381]:
```suggestion
			const listItem = data.modelRange.start.nodeAfter || data.modelRange.end.nodeBefore;
```

**Comment by oleq** [comment]
You merged this while I was writing a comment to ask for ETA :D Perfect timing.

**Pull request merged**